crates/codegen/parser/runtime/src/templates/queries.rs.jinja2

```diff
@@ -0,0 +1,56 @@
+use std::sync::Once;
+use crate::generated::query::Query;
+
+mod __query_kind {
+    #![allow(unreachable_code)]
+    #[cfg(feature = "slang_napi_interfaces")]
+    use napi_derive::napi;
+    #[derive(
+        Debug,
+        Eq,
+        Hash,
+        Ord,
+        PartialEq,
+        PartialOrd,
+        serde::Serialize,
+        strum_macros::AsRefStr,
+        strum_macros::Display,
+        strum_macros::EnumString,
+    )]
+    #[cfg_attr(feature = "slang_napi_interfaces", /* derives `Clone` and `Copy` */ napi(string_enum, namespace = "queries"))]
+    #[cfg_attr(not(feature = "slang_napi_interfaces"), derive(Clone, Copy))]
+    pub enum QueryKind {
+        {%- for name, _ in queries -%}
+            {# variant.documentation | indent(prefix = "/// ", first = true, blank = true) #}
+            {{ name | pascal_case }},
+        {%- endfor -%}
+    }
+}
+pub use __query_kind::QueryKind;
+
+static mut QUERIES: Vec<Query> = Vec::new();
+static INIT: Once = Once::new();
+
+pub struct Queries;
+
+impl std::ops::Index<QueryKind> for Queries {
+    type Output = Query;
+
+    fn index(&self, kind: QueryKind) -> &Self::Output {
+        unsafe {
```

Comment:
I suggest using `std::sync::OnceLock`, instead of unsafe code.

Reply:
This is straight from the Rust docs ... F2F?

Reply:
Not a major concern, but I suggest avoiding `unsafe` anywhere unless we really need it.
I suspect these specific docs might have been there before adding `OnceLock`.
Here is another example from: https://doc.rust-lang.org/std/sync/struct.OnceLock.html

```rust
struct DeepThought {
    answer: String,
}

fn computation() -> &'static DeepThought {
    static COMPUTATION: OnceLock<DeepThought> = OnceLock::new();
    COMPUTATION.get_or_init(|| DeepThought::new())
}

let _ = computation().answer;
```

Happy to F2F if I missed anything.

Reply:
obsoleted